openmc/mgxs_library.py

```diff
@@ -67,22 +71,7 @@ class XSdata(object):
         is "angle".
     num_polar : int
         Number of equal width angular bins that the polar angular domain is
-        subdivided into. This only applies when :attr:`XSdata.representation`
-        is "angle".
-    use_chi : bool
-        Whether or not a chi vector or nu-fission matrix was used.
-    vector_shape : iterable of int
```

Comment:
I just found that with delayed groups the lines between get blurry on what is a matrix shape. Whereas before vector shape noted just `groups` and matrix noted `groups x groups`, with delayed groups there would need to be shape terms for `energy_groups`, `energy_groups x energy_groups`, `delayed_groups`, `delayed_groups x energy_groups`, and `delayed_groups x energy_groups x energy_groups`. I'm go another PR on deck for a surface mgxs class which would add shapes for `mesh_surfaces` and `energy_groups x mesh_surfaces`. And then there is the `pn_matrix_shape` as well.

I was just thinking it is starting to get bloated with several different shape presets and might be best to just do them individually. How about a dictionary with all the different shape options and then selecting the ones available for the particular mgxs of interest when they are needed?